### PR TITLE
Issuer can view connection details

### PR DESCRIPTION
**Pull request by adrifdez**
Resolves PID 436

### CODE REVIEW

File: ui/src/components/connections/ConnectionDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+                  <Space direction="vertical" size="middle">
+                    <Row align="middle" justify="space-between">
+                      <Card.Meta title="Connection" />
+                      <Button danger icon={<IconTrash />} onClick={onClickToImplement} type="text">
+                        Delete connection
+                      </Button>
+                    </Row>
+                    <Card className="background-grey">
+                      <Row justify="space-between">
+                        <Typography.Text type="secondary">{IDENTIFIER}</Typography.Text>
```
[the remark this answers is not in the document]
Comment (adrifdez) [reply ui/src/components/connections/ConnectionDetails.tsx:101]: Yes in connections but I forgot to add it there or remove it when updating from the other branch

File: ui/src/utils/constants.ts [hunk 2]
```diff
@@ -46,6 +47,8 @@ export const ROOT_PATH = "/";
 
 export const SIDER_WIDTH = 320;
 
+export const TO_DEVELOP = "To develop";
```
Comment (cneuro) [inline ui/src/utils/constants.ts:50]: This has never been necessary, and I would urge us not to start doing this, for the following reasons;

1. It's a useless constant in production
2. It's phrased awkwardly
3. It complicates development by having to keep track of a WIP flag
4. There are plenty of unimplemented parts that do not have this flag and adding it only in a few places makes things arbitrary and inconsistent
5. This is a FOSS repo, and having non-functional bits like this makes development confusing and inconsistent

What's wrong with just omitting this altogether? If it's not implemented, it simply isn't, functions and notices don't need to say this explicitly. @amonsosanz, thoughts?

File: ui/src/components/connections/ConnectionDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+import { useParams } from "react-router-dom";
+
+import { APIError } from "src/adapters/api";
+import { Connection, getConnection } from "src/adapters/api/connections";
+import { ReactComponent as IconCheckMark } from "src/assets/icons/check.svg";
+import { ReactComponent as IconCopy } from "src/assets/icons/copy-01.svg";
+import { ReactComponent as IconCreditCardPlus } from "src/assets/icons/credit-card-plus.svg";
+import { ReactComponent as IconCreditCardRefresh } from "src/assets/icons/credit-card-refresh.svg";
+import { ReactComponent as IconTrash } from "src/assets/icons/trash-01.svg";
+
```
Comment (amonsosanz) [inline ui/src/components/connections/ConnectionDetails.tsx:12]: extra line

File: ui/src/utils/constants.ts (hunk 2 above)
Comment (amonsosanz) [reply ui/src/utils/constants.ts]: I'm ok removing this.

File: ui/src/components/connections/ConnectionDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+              }
+              case "successful":
+              case "reloading": {
+                return (
+                  <Space direction="vertical" size="middle">
+                    <Row align="middle" justify="space-between">
+                      <Card.Meta title="Connection" />
+                      <Button danger icon={<IconTrash />} type="text">
+                        Delete connection
+                      </Button>
```
Comment (amonsosanz) [inline ui/src/components/connections/ConnectionDetails.tsx:103]: Shouldn't these buttons be consistently styled?

delete-connection

Comment (cneuro) [reply ui/src/components/connections/ConnectionDetails.tsx:103]: @amonsosanz the dropdown is a link as opposed to the button. It's correct within the system, they are different modalities.

**Review by cneuro — APPROVED** [review]
Great work @adrifdez

File: ui/src/components/connections/ConnectionDetails.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+            data: response.data,
+            status: "successful",
+          });
+        } else {
+          if (!isAbortedError(response.error)) {
+            setConnection({ error: response.error, status: "failed" });
+          }
+        }
+      } else {
+        setConnection({ error: { message: "No ConnectionID" }, status: "failed" });
```
Comment (amonsosanz) [inline ui/src/components/connections/ConnectionDetails.tsx:60]: Can this happen?

File: ui/src/routes.ts [hunk 6]
```diff
@@ -19,6 +20,10 @@ type Routes = Record<
 >;
 
 export const ROUTES: Routes = {
+  connectionDetails: {
+    layout: "sider",
+    path: "/connection/:connectionID",
```
Comment (amonsosanz) [inline ui/src/routes.ts:25]: This path should be plural: `/connections/:connectionID`

File: ui/src/components/connections/ConnectionDetails.tsx (hunk 4 above)
Comment (amonsosanz) [reply ui/src/components/connections/ConnectionDetails.tsx:103]: ok...

File: ui/src/components/connections/ConnectionDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+                          {obfuscateDID(connection.data.userID)}
+                        </Typography.Text>
+                      </Row>
+                      <Row justify="space-between">
+                        <Typography.Text type="secondary">Creation date</Typography.Text>
+                        <Typography.Text>
+                          {formatDate(connection.data.createdAt, true)}
+                        </Typography.Text>
+                      </Row>
+                    </Card>
```
Comment (amonsosanz) [inline ui/src/components/connections/ConnectionDetails.tsx:121]:
```diff
diff --git a/ui/src/components/connections/ConnectionDetails.tsx b/ui/src/components/connections/ConnectionDetails.tsx
index 3338a44..5c282b4 100644
--- a/ui/src/components/connections/ConnectionDetails.tsx
+++ b/ui/src/components/connections/ConnectionDetails.tsx
@@ -2,10 +2,9 @@ import { Avatar, Button, Card, Row, Space, Tag, Typography } from "antd";
 import { useCallback, useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 
+import { Detail } from "../schemas/Detail";
 import { APIError } from "src/adapters/api";
 import { Connection, getConnection } from "src/adapters/api/connections";
-import { ReactComponent as IconCheckMark } from "src/assets/icons/check.svg";
-import { ReactComponent as IconCopy } from "src/assets/icons/copy-01.svg";
 import { ReactComponent as IconCreditCardPlus } from "src/assets/icons/credit-card-plus.svg";
 import { ReactComponent as IconCreditCardRefresh } from "src/assets/icons/credit-card-refresh.svg";
 import { ReactComponent as IconTrash } from "src/assets/icons/trash-01.svg";
@@ -103,23 +102,15 @@ export function ConnectionDetails() {
                       </Button>
                     </Row>
                     <Card className="background-grey">
-                      <Row justify="space-between">
-                        <Typography.Text type="secondary">{IDENTIFIER}</Typography.Text>
-                        <Typography.Text
-                          copyable={{
-                            icon: [<IconCopy key={0} />, <IconCheckMark key={1} />],
-                            text: connection.data.userID,
-                          }}
-                        >
-                          {obfuscateDID(connection.data.userID)}
-                        </Typography.Text>
-                      </Row>
-                      <Row justify="space-between">
-                        <Typography.Text type="secondary">Creation date</Typography.Text>
-                        <Typography.Text>
-                          {formatDate(connection.data.createdAt, true)}
-                        </Typography.Text>
-                      </Row>
+                      <Detail
+                        copyable
+                        data={obfuscateDID(connection.data.userID)}
+                        label={IDENTIFIER}
+                      />
+                      <Detail
+                        data={formatDate(connection.data.createdAt, true)}
+                        label="Creation date"
+                      />
                     </Card>
```

File: ui/src/components/connections/ConnectionDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+
+                <Typography.Text type="secondary">
+                  Credentials for this connection will be listed here.
+                </Typography.Text>
+              </>
+            }
+            isLoading={isAsyncTaskStarting(connection)}
+            onSearch={() => null}
+            query=""
+            searchPlaceholder="Search credentials, attributes..."
```
Comment (amonsosanz) [inline ui/src/components/connections/ConnectionDetails.tsx:144]: `Search connections, credentials...`

Comment (amonsosanz) [reply ui/src/components/connections/ConnectionDetails.tsx:135]: Should we then [amend the design](https://www.figma.com/file/BE3tUbvtSB2EqmJsybpeFC/Polygon-ID-Platform?node-id=4749-162667&t=KoTPRUTLBtQ183Ui-0) @Figmintojas?

**Review by amonsosanz — APPROVED** [review]

**Pull request merged**